sodium/crypto_sign.pony

```diff
@@ -47,7 +56,15 @@ primitive CryptoSign
     let pk_size = public_key_size(); let pk = _make_buffer(pk_size)
     if 0 != @crypto_sign_keypair[_Int](pk.cstring(), sk.cstring()) then error end
     (CryptoSignSecretKey(consume sk), CryptoSignPublicKey(consume pk))
-  
+
+  fun tag seed_keypair(seed: CryptoSignSeed): (CryptoSignSecretKey, CryptoSignPublicKey)? =>
+    let sk_size = secret_key_size(); let sk = _make_buffer(sk_size)
```

Comment:
Note that the other methods which accept a size-constrained parameter will check the parameter size here by calling `is_valid()`.  For example, see https://github.com/jemc/pony-sodium/pull/1/files#diff-b46b59da072f172e8dc5639614d00517R69

This will cause the operation to `error` if the size of the seed doesn't match the expected size (which you said is 32 bytes).